R/repeated_measures_d.R

```diff
@@ -0,0 +1,302 @@
+    out$CI <- ci
+    ci.level <- .adjust_ci(ci, alternative)
+
+    alpha <- 1 - ci.level
+    probs <- c(alpha / 2, 1 - alpha / 2)
+    qs <- stats::qnorm(probs)
+
+    confint <- out[["d"]] + qs * values[["se"]]
+    out$CI_low <- confint[1]
+    out$CI_high <- confint[2]
+
+    ci_method <- list(method = "normal")
+    out <- .limit_ci(out, alternative, -Inf, Inf)
+  } else {
+    ci_method <- alternative <- NULL
+  }
+
+
+  if (adjust) {
+    J <- .J(values[["df"]])
+
+    out[, colnames(out) %in% c("d", "CI_low", "CI_high")] <-
+      out[, colnames(out) %in% c("d", "CI_low", "CI_high")] * J
+
+    attr(out, "table_footer") <- "Adjusted for small sample bias."
+  }
+
+  # rename column to method
+  colnames(out)[1] <- switch(method,
+    d = "Cohens_d",
+    b = "Beckers_d",
+    paste0("d_", method)
+  )
+
+  class(out) <- c("effectsize_difference", "effectsize_table", "see_effectsize_table", class(out))
+  .someattributes(out) <- .nlist(
+    mu, ci, ci_method, alternative,
+    approximate = FALSE
+  )
+  return(out)
+}
+
+#' @rdname repeated_measures_d
+#' @export
+rm_d <- repeated_measures_d
+
+#' @keywords internal
+.paired_d <- function(data, mu, method) {
+  x <- data[["x"]]
+  y <- data[["y"]]
+
+  m <- mean(x - y)
+  n <- length(x)
+  df <- n - 1
+  r <- stats::cor(x, y)
+
+  if (method == "rm") {
+    f <- 2 * (1 - r)
+
+    s <- stats::sd(x - y) / sqrt(f)
+    d <- (m - mu) / s
+
+    # Cooper et al., 2009, eq 12.21
+    se <- sqrt(((1 / n) + (d^2) / (2 * n)) * f)
+  } else if (method == "av") {
+    s <- sqrt((stats::var(x) + stats::var(y)) / 2)
+    d <- (m - mu) / s
+
+    # Algina & Keselman, 2003, eq 4 (from Bird 2002?)
+    se <- sqrt(stats::var(x - y) / (s^2 * n))
+  } else if (method == "z") {
+    s <- stats::sd(x - y)
+    d <- (m - mu) / s
+
+    # Hedges and Olkin, 1985, page 86, eq 14
+    se <- sqrt((1 / n) + (d^2) / (2 * n))
+  } else if (method == "b") {
+    s <- stats::sd(y)
+    d <- (m - mu) / s
+
+    # Becker 1988, eq. 6
+    se <- sqrt((2 * (1 - r) / n) + (d^2) / (2 * n))
+  }
+
+  .nlist(d, se, df)
+}
+
+#' @keywords internal
+.replication_d <- function(data, mu, method) {
+  if (method == "r") {
+    # for r - need to make sure there are replications!
+    cell_ns <- tapply(data[[1]], data[3:2], function(v) length(stats::na.omit(v)))
+    all(cell_ns > 1L)
+  }
+
+  mod <- suppressWarnings(
+    stats::aov(y ~ condition + Error(id / condition),
+      data = data,
+      contrasts = list(condition = contr.treatment)
+    )
+  )
+  m <- -unname(coef(mod[["id:condition"]]))
+  m_V <- unname(vcov(mod[["id:condition"]])[1])
+
+  pars <- parameters::model_parameters(mod)
+
+  if (method == "d") {
+    e <- as.data.frame(pars[pars$Parameter == "Residuals", ])
+  } else if (method == "r") {
+    e <- as.data.frame(pars[pars$Group == "Within", ])
+  }
+
+  s <- sqrt(sum(e[["Sum_Squares"]]) / sum(e[["df"]]))
+  df <- sum(e[["df"]])
+
+  d <- (m - mu) / s
+  se <- sqrt(
+    (df / (df - 2)) * (m_V / (s^2)) +
```

Comment:
so I think this `(df/(df-2))` term is a small sample correction factor that seems to be applied broadly (i have seen it in other SE equations too). It probably does not matter all that much, but it might be good to stay consistent

Reply:
Yeah, but I don't feel comfortable making that change myself 😅😳